### DISCUSSION TITLE
@keystone-6/cloudinary support to other files types upload than images

### DISCUSSION THREAD

**Discussion by yangricardo** [keystonejs/keystone#7391] [Feature requests]
Hi, 
As the Cloudinary supports other media types upload, like PDFs, Word and any other, would be nice to have this capabilty on this plugin. I am new to Keystone 6, and i am studying on how to do that with a custom fields, based on this project. I had an idea that the only thing is blocking is on the [Field.tsx#L166](https://github.com/keystonejs/keystone/blob/main/packages/cloudinary/src/views/Field.tsx#L166), that do some check based on type / mimetype....
Another point of the code to update would be the render, that could just be a link to download the file.
Instead of deny other file upload types, these could be tagged as type "image" | "file", in order to handle how to preview the uploaded file.
Regards.

**Comment by molomby** [comment]
Hey yangricardo, you're right that this would potentially not be a massive change to the field type but it's not a change we're interested in making to this core package. We have some file handling functionality due for released soon that will likely solve this better for many use case, so making a change like this is pretty low priority.

That said, it's definitely something you could fix in your own project with a custom field type. You should more or less, just be able to copy the `packages/cloudinary` directory into your project then make the tweaks you want. If I were you I'd start with the [custom field example](https://github.com/keystonejs/keystone/tree/main/examples/custom-field) project. The [custom fields guide](https://keystonejs.com/docs/guides/custom-fields) should have the other details you need.

**Reply by yangricardo** [reply]
Hey @molomby , thanks for the answer. Just after i've sent this future request i have tried to reconstruct the packages/cloudinary as a custom field, doing some small changes, and was kind successfull, by doing some changes as enable other file types than images. 
At this point i could upload the file, but the cloudinary free tier blocked the public url generation, that is, i could just get the file by accessing their platform. To this, i think it´s correct your choice of just handle images. Maybe it should be a warning on your docs.
But to the experience of the field usage, i would recomend that the cloudinary key arguments could be an environment variable, like `CLOUDINARY_SECRETKEY`. This would shorter the way this component is used, as the developer does not need to put these key args on every field definition for a list schema.

Also, there is anything to an AWS S3 integration? I figure out that it already exists on keystone 5, but not on 6. I will try to develop a custom field to handle AWS S3 uploads.

**Reply by aaronpowell** [reply]
If you're after S3, check out the contrib field - https://github.com/keystonejs-contrib/k6-contrib/tree/main/packages/fields-s3images

Also, keep an eye on https://github.com/keystonejs/keystone/pull/7070

**Reply by WenLonG12345** [reply]
HI, I was beginner in JS world. Mind to ask if how can I copy the `package/cloudinary` into my project? I had created a `cloudinary` folder into keystonejs project, move all the core logic inside the folder and add dependecies in `package.json` But it failed to compile. Any idea?
